From 5bb3161566b3540e6909e585675eab225dcb994a Mon Sep 17 00:00:00 2001
From: Ulf Hermann <ulf.hermann@qt.io>
Date: Tue, 26 Sep 2023 12:36:09 +0200
Subject: QtQml: Re-add pthread_attr_init() to stackPropertiesGeneric()

Before calling pthread_attr_get_np(), as opposed to
pthread_getattr_np(), we do need to call pthread_attr_init(). Both the
FreeBSD and the NetBSD manpages tell us to do so.

Amends commit 9f4aeeabb982cfc4306c9d350dbb68f64914fb32.

Pick-to: 6.6 6.5
Fixes: QTBUG-117513
Change-Id: Ic851ba2ffcf13d268b3a53d926cb92f7bed7a3d9
(cherry picked from commit 94e0bb6de357cb5e1fca32a4107363d95346222e)
---
 src/qml/memory/qv4stacklimits.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git x/qtdeclarative/src/qml/memory/qv4stacklimits.cpp y/qtdeclarative/src/qml/memory/qv4stacklimits.cpp
index 68ef7a366a..429520e527 100644
--- x/qtdeclarative/src/qml/memory/qv4stacklimits.cpp
+++ y/qtdeclarative/src/qml/memory/qv4stacklimits.cpp
@@ -235,6 +235,7 @@ StackProperties stackPropertiesGeneric(qsizetype stackSize = 0)
     pthread_t thread = pthread_self();
     pthread_attr_t sattr;
 #  if defined(PTHREAD_NP_H) || defined(_PTHREAD_NP_H_) || defined(Q_OS_NETBSD)
+    pthread_attr_init(&sattr);
     pthread_attr_get_np(thread, &sattr);
 #  else
     pthread_getattr_np(thread, &sattr);
